forage-android/src/main/java/com/joinforage/forage/android/network/model/ForageApiResponse.kt

```diff
@@ -5,15 +5,29 @@ import org.json.JSONObject
 sealed class ForageApiResponse<out T> {
     data class Success<out T>(val data: T) : ForageApiResponse<T>()
 
-    data class Failure(val errors: List<ForageError>) : ForageApiResponse<Nothing>()
+    data class Failure(val errors: List<ForageError>) : ForageApiResponse<Nothing>() {
+        companion object {
+            fun fromSQSError(sqsError: SQSError): Failure {
+                val forageError = ForageError(
+                    sqsError.statusCode,
+                    sqsError.forageCode,
+                    sqsError.message,
+                    sqsError.details
+                )
+                return Failure(listOf(forageError))
+            }
+        }
+    }
 }
 
 data class ForageError(
     val httpStatusCode: Int,
     val code: String,
-    val message: String
+    val message: String,
+    val details: JSONObject? = null,
```

Comment:
Can we make the client-facing type stricter than JSONObject? and unmarshall the response into a typed `cashBalance` and `snapBalance`

Reply:
yup! done in the latest commit. Once I flushed out the test suite a bit more I started getting really loud errors from just using JSONObject so good call